letta/schemas/tool_rule.py

```diff
@@ -21,6 +21,17 @@ class ChildToolRule(BaseToolRule):
     children: List[str] = Field(..., description="The children tools that can be invoked.")
 
 
+class ConditionalToolRule(BaseToolRule):
+    """
+    A ToolRule that conditionally maps to different child tools based on the output.
+    """
+    type: ToolRuleType = ToolRuleType.conditional
+    default_child: str = Field(..., description="The default child tool to be called")
+    child_output_mapping: Dict[Union[bool, str, int], str] = Field(..., description="The output case to check for mapping")
+    children: List[str] = Field(..., description="The child tool to call when output matches the case")
+    throw_error: bool = Field(default=False, description="Whether to throw an error when output doesn't match any case")
```

Comment:
maybe rename this to `require_output_mapping` to make more clear what this means

Reply:
yeah this is repetitive. removed `children`